File: GO/awesomeProject/example_0504/BubbleSort.go
package main

import "fmt"

func main() {
	arr := []int{1, 7, 9, 8, 60, 70, 54, 21, 36, 69}
	BubbleSort(arr)
	fmt.Println("The sort array is ", arr)
}
func BubbleSort(arr []int) {
	for i := 0; i < len(arr); i++ {
		for j := 0; j < len(arr)-i-1; j++ {
			if arr[j] > arr[j+1] {
				tmp := arr[j]
				arr[j] = arr[j+1]
				arr[j+1] = tmp
			}
		}
	}
}

File: java/Study.java/src/dataframe/sort/BubbleSort.java
package dataframe.sort;
//冒泡排序：交换排序
public class BubbleSort {
    public static void bubbleSort(int [] a){
        //外层循环控制比较的次数
        for(int i = 0;i < a.length-1;i++){
            //内层循环控制比较元素到达的位置
            for(int j = 0;j<a.length-i-1;i++){
                if(a[j] >a[j+1]){
                    //前面比后面大就交换位置
                    int temp = a[j];
                    a[j]=a[j+1];
                    a[j+1]=temp;
                }
            }
        }
    }
}

